lib/src/builder.dart

```diff
@@ -2051,7 +2051,20 @@ class _MockClassInfo {
 
   T _withTypeParameters<T>(Iterable<TypeParameterElement> typeParameters,
       T Function(Iterable<TypeReference>, Iterable<TypeReference>) body) {
-    final typeVars = {for (final t in typeParameters) t: _newTypeVar(t)};
+    final Map<TypeParameterElement, String> typeVars = {};
```

Comment:
As I wrote on the bug, I think it would be more logical to fix the equality on the analyzer side, if possible. Then, this won't be needed.

But even as a workaround, it has a few problems:
1. It won't catch cases with more than one level of wrapping, like `T extends Foo<Bar<T>>`.
2. It will create a wrong mapping for _other_ type variables appearing in the bound, like
    ```dart
    class C<T> {
      X m<X extends Iterable<T>>() => [];
    }
    ```